src/app/components/calculator/calculator.component.ts

```diff
@@ -0,0 +1,38 @@
+import { Component, OnInit } from '@angular/core';
+import { Calculator } from 'src/app/classes/calculator';
+import { History } from 'src/app/classes/history';
+import { CoursesService } from 'src/app/services/courses.service';
+
+@Component({
+  selector: 'app-calculator',
+  templateUrl: './calculator.component.html',
+  styleUrls: ['./calculator.component.scss']
+})
+export class CalculatorComponent implements OnInit {
+  totalHours: number | undefined;
+  hoursPerWeek: number | undefined;
+
```

Comment:
adiciona uma propriedade privada pra lidar com os .subscribe()
```suggestion
private openSubscriptions: Subscription[] = []
```